modules/tinymce/src/core/main/ts/api/EditorOptions.ts

```diff
@@ -0,0 +1,277 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Arr, Obj, Optional, Strings, Type } from '@ephox/katamari';
+
+import Editor from './Editor';
+import { EditorOptions, EditorSettings } from './OptionTypes';
+
+interface ProcessorSuccess<T> {
+  valid: true;
+  value: T;
+}
+
+interface ProcessorError {
+  valid: false;
+  message: string;
+}
+
+type SimpleProcessor<T> = (value: T) => boolean;
+type Processor<T, U> = (value: T) => ProcessorSuccess<U> | ProcessorError;
+
+export interface BuiltInOptionTypeMap {
+  'string': string;
+  'number': number;
+  'boolean': boolean;
+  'string[]': string[];
+  'array': any[];
+  'function': Function;
+  'object': any;
+}
+
+export type BuiltInOptionType = keyof BuiltInOptionTypeMap;
+
+interface BaseOptionSpec {
+  immutable?: boolean;
+  deprecated?: boolean;
+  docsUrl?: string;
+}
+
+export interface BuiltInOptionSpec<K extends BuiltInOptionType> extends BaseOptionSpec {
+  processor: K;
+  default?: BuiltInOptionTypeMap[K];
+}
+
+export interface SimpleOptionSpec<T> extends BaseOptionSpec {
+  processor: SimpleProcessor<T>;
+  default?: T;
+}
+
+export interface OptionSpec<T, U> extends BaseOptionSpec {
+  processor: Processor<T, U>;
+  default?: T;
+}
+
+/**
+ * TinyMCE Options API.
+ *
+ * @class tinymce.editor.Options
+ */
+
+export interface Options {
+  /**
+   * Register a new option that can be used within TinyMCE.
+   *
+   * @method register
+   * @param {String} name Name of an option.
```

Comment:
```suggestion
   * @param {String} name Name of the option.
```

Reply:
I don't like `the` here as it makes it sound like it's a specific option name you need to pass, not any option name.

Reply:
I think we should probably have @tylerkelly13 run through all this at some point tbh

Reply:
Yeah we need him to review this at some stage. I was going to add him to the reviewers list when I convert it from a draft.

Reply:
I prefer "Name of the option", we refer to it as the option in the spec doc string. it's the option you're registering, not a specific, ambiguous option

Reply:
```suggestion
   * @param {String} name Name of the option.
```